src/emqttd_auth_clientid.erl

```diff
@@ -40,87 +30,71 @@
 
 -record(?AUTH_CLIENTID_TAB, {client_id, ipaddr, password}).
 
-%%%=============================================================================
-%%% API
-%%%=============================================================================
+%%--------------------------------------------------------------------
+%% API
+%%--------------------------------------------------------------------
 
-%%------------------------------------------------------------------------------
 %% @doc Add clientid
-%% @end
-%%------------------------------------------------------------------------------
 -spec add_clientid(binary()) -> {atomic, ok} | {aborted, any()}.
 add_clientid(ClientId) when is_binary(ClientId) ->
     R = #mqtt_auth_clientid{client_id = ClientId},
     mnesia:transaction(fun mnesia:write/1, [R]).
 
-%%------------------------------------------------------------------------------
 %% @doc Add clientid with password
-%% @end
-%%------------------------------------------------------------------------------
 -spec add_clientid(binary(), binary()) -> {atomic, ok} | {aborted, any()}.
 add_clientid(ClientId, Password) ->
     R = #mqtt_auth_clientid{client_id = ClientId, password = Password},
     mnesia:transaction(fun mnesia:write/1, [R]).
 
-%%------------------------------------------------------------------------------
 %% @doc Lookup clientid
-%% @end
-%%------------------------------------------------------------------------------
--spec lookup_clientid(binary()) -> list().
+-spec lookup_clientid(binary()) -> list(#mqtt_auth_clientid{}).
```

Comment:
The spec in line 50 uses a record, please define a type for the record and use that instead.